src/libraries/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -588,52 +588,71 @@ private static unsafe void Ssse3Decode(ref byte* srcBytes, ref byte* destBytes,
+                Vector128<byte> hi = SimdShuffle(lutHi, hiNibbles, mask8F);
+                Vector128<byte> lo = SimdShuffle(lutLo, str, mask8F);
 
                 // Check for invalid input: if any "and" values from lo and hi are not zero,
                 // fall back on bytewise code to do error checking and reporting:
-                if (Sse2.MoveMask(Sse2.CompareGreaterThan(Sse2.And(lo, hi), zero)) != 0)
+                if ((lo & hi) != Vector128<byte>.Zero)
                     break;
 
-                Vector128<sbyte> eq2F = Sse2.CompareEqual(str, mask2F);
-                Vector128<sbyte> shift = Ssse3.Shuffle(lutShift, Sse2.Add(eq2F, hiNibbles));
+                Vector128<byte> eq2F = Vector128.Equals(str, mask2F);
+                Vector128<byte> shift = SimdShuffle(lutShift.AsByte(), (eq2F + hiNibbles), mask8F);
 
                 // Now simply add the delta values to the input:
-                str = Sse2.Add(str, shift);
+                str += shift;
 
                 // in, bits, upper case are most significant bits, lower case are least significant bits
                 // 00llllll 00kkkkLL 00jjKKKK 00JJJJJJ
                 // 00iiiiii 00hhhhII 00ggHHHH 00GGGGGG
                 // 00ffffff 00eeeeFF 00ddEEEE 00DDDDDD
                 // 00cccccc 00bbbbCC 00aaBBBB 00AAAAAA
 
-                Vector128<short> merge_ab_and_bc = Ssse3.MultiplyAddAdjacent(str.AsByte(), mergeConstant0);
+                Vector128<short> merge_ab_and_bc;
+                if (Ssse3.IsSupported)
+                {
+                    merge_ab_and_bc = Ssse3.MultiplyAddAdjacent(str.AsByte(), mergeConstant0.AsSByte());
+                }
+                else
+                {
+                    Vector128<ushort> evens = AdvSimd.ShiftLeftLogicalWideningLower(AdvSimd.Arm64.UnzipEven(str, one).GetLower(), 6);
+                    Vector128<ushort> odds = AdvSimd.Arm64.TransposeOdd(str, Vector128<byte>.Zero).AsUInt16();
+                    merge_ab_and_bc = Vector128.Add(evens, odds).AsInt16();
+                }
                 // 0000kkkk LLllllll 0000JJJJ JJjjKKKK
                 // 0000hhhh IIiiiiii 0000GGGG GGggHHHH
                 // 0000eeee FFffffff 0000DDDD DDddEEEE
                 // 0000bbbb CCcccccc 0000AAAA AAaaBBBB
 
-                Vector128<int> output = Sse2.MultiplyAddAdjacent(merge_ab_and_bc, mergeConstant1);
+                Vector128<int> output;
+                if (Ssse3.IsSupported)
+                {
+                    output = Sse2.MultiplyAddAdjacent(merge_ab_and_bc, mergeConstant1);
+                }
+                else
+                {
+                    Vector128<int> ievens = AdvSimd.ShiftLeftLogicalWideningLower(AdvSimd.Arm64.UnzipEven(merge_ab_and_bc, one.AsInt16()).GetLower(), 12);
+                    Vector128<int> iodds = AdvSimd.Arm64.TransposeOdd(merge_ab_and_bc, Vector128<short>.Zero).AsInt32();
+                    output = Vector128.Add(ievens, iodds).AsInt32();
```

Comment:
super nit: could be `(ievens + iodds).AsInt32()` , but we can do it in later PR.